game/src/pregame/mod.rs

```diff
@@ -1,374 +1,46 @@
-                .text("See full credits")
-                .build_def(ctx)
-                .centered_horiz(),
-        ];
-
-        Box::new(About {
-            panel: Panel::new_builder(Widget::custom_col(col))
-                .exact_size_percent(90, 85)
-                .build_custom(ctx),
-        })
-    }
-}
-
-impl State<App> for About {
-    fn event(&mut self, ctx: &mut EventCtx, _: &mut App) -> Transition {
-        if let Outcome::Clicked(x) = self.panel.event(ctx) {
-            match x.as_ref() {
-                "back" => {
-                    return Transition::Pop;
-                }
-                "See full credits" => {
-                    open_browser("https://github.com/a-b-street/abstreet#credits");
-                }
-                _ => unreachable!(),
-            }
-        }
-
-        Transition::Keep
-    }
-
-    fn draw_baselayer(&self) -> DrawBaselayer {
-        DrawBaselayer::Custom
-    }
-
-    fn draw(&self, g: &mut GfxCtx, app: &App) {
-        g.clear(app.cs.dialog_bg);
-        self.panel.draw(g);
-    }
-}
-
-struct Screensaver {
-    line: Line,
-    started: Instant,
-}
-
-impl Screensaver {
-    fn bounce(ctx: &mut EventCtx, app: &mut App, rng: &mut XorShiftRng) -> Screensaver {
-        let at = ctx.canvas.center_to_map_pt();
-        let bounds = app.primary.map.get_bounds();
-        let line = loop {
-            let goto = Pt2D::new(
-                rng.gen_range(0.0..bounds.max_x),
-                rng.gen_range(0.0..bounds.max_y),
-            );
-            if let Some(l) = Line::new(at, goto) {
-                break l;
-            }
-        };
-        ctx.canvas.cam_zoom = 10.0;
-
-        Screensaver {
-            line,
-            started: Instant::now(),
-        }
-    }
-
-    fn update(&mut self, rng: &mut XorShiftRng, ctx: &mut EventCtx, app: &mut App) {
-        const SIM_SPEED: f64 = 3.0;
-        const PAN_SPEED: Speed = Speed::const_meters_per_second(20.0);
-
-        if let Some(dt) = ctx.input.nonblocking_is_update_event() {
-            ctx.input.use_update_event();
-            if let Some(pt) = self
-                .line
-                .dist_along(Duration::realtime_elapsed(self.started) * PAN_SPEED)
-            {
-                ctx.canvas.center_on_map_pt(pt);
-            } else {
-                *self = Screensaver::bounce(ctx, app, rng);
-            }
-            app.primary.sim.time_limited_step(
-                &app.primary.map,
-                SIM_SPEED * dt,
-                Duration::seconds(0.033),
-                &mut app.primary.sim_cb,
-            );
-        }
+    pub fn new_state(ctx: &mut EventCtx, app: &mut App) -> Box<dyn State<App>> {
+        map_gui::tools::TitleScreen::new_state(
+            ctx,
+            app,
+            map_gui::tools::Executable::ABStreet,
+            Box::new(enter_state),
+        )
+    }
+}
+
+fn enter_state(ctx: &mut EventCtx, app: &mut App, args: Vec<&str>) -> Box<dyn State<App>> {
```

Comment:
This callback is used by the generic title screen code. It's kind of unfortunate that this repeats some of the logic from `game/src/lib.rs`. I'll think about if there's a way to consolidate sanely.